scripts/new_version

```diff
@@ -44,9 +54,14 @@ def new_version(new_version_string=''):
         raise ValueError(
             'There must be at least one new change in CHANGELIST.md')
 
-    with open(VERSION_FILE, 'w') as fp:
-        fp.write(new_version_string)
-        fp.write('\n')
+    with open(INIT_FILE, 'r+', encoding='utf8') as fp:
+        init = fp.read()
+        sub = re.subn(r"__version__\s*=\s*\\*[\'\"]([^\'\"]*)\\*[\'\"]",
```

Comment:
There has to be a better way than this.  I'm good at regular expression and I have absolutely no idea what it does.

More, regular expressions have proven time and again to be fragile - that is, liable to break if you make otherwise correct changes.

See also:  http://regex.info/blog/2006-09-15/247

Reply:
Agree'd.
This regex looks for a line including `__version__`, has lazy space matching `\s*`, takes care of back slashes in the `init` stream, and matches any character within the single quotation marks (for versioning)

I've used this in the past before with no repercussion, but as always, YMMV.